recipes/ios-cmake/all/conanfile.py

```diff
@@ -29,121 +51,126 @@ class IosCMakeConan(ConanFile):
+        elif self.settings.os == "watchOS":
+            if self.settings.arch == "x86_64":
                 return "SIMULATOR_WATCHOS"
             else:
                 return "WATCHOS"
-        elif os == "tvOS":
-            if arch == "x86_64":
+        elif self.settings.os == "tvOS":
+            if self.settings.arch == "x86_64":
                 return "TVOS"
             else:
                 return "SIMULATOR_TVOS"
-        raise ConanInvalidConfiguration("Can not guess toolchain_target. Please set the option explicit (or check our os settings)")
+        return None
+
+    def config_options(self):
+        self.options.toolchain_target = self._default_toolchain_target or "auto"
+
+    def export_sources(self):
+        copy(self, "cmake-wrapper", self.recipe_folder, self.export_sources_folder)
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def validate(self):
+        if not is_apple_os(self):
+            raise ConanInvalidConfiguration("This package only supports Apple operating systems")
+        if self.options.toolchain_target == "auto":
+            raise ConanInvalidConfiguration("Cannot guess toolchain_target. Please set the option explicitly.")
 
+    def package_id(self):
+        self.info.clear()
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        os.rename("ios-cmake-{}".format(self.version), self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def build(self):
-        pass # there is nothing to build
+        pass  # there is nothing to build
+
+    @staticmethod
+    def _chmod_plus_x(filename):
+        if os.name == "posix":
+            os.chmod(filename, os.stat(filename).st_mode | 0o111)
 
     def package(self):
-        self.copy("cmake-wrapper", dst="bin")
-        self.copy("ios.toolchain.cmake",
-                    src=self._source_subfolder,
-                    dst=os.path.join("lib", "cmake", "ios-cmake"),
-                    keep_path=False)
+        copy(self, "cmake-wrapper",
+             src=self.export_sources_folder,
+             dst=os.path.join(self.package_folder, "bin"))
+        copy(self, "ios.toolchain.cmake",
+             src=self.source_folder,
+             dst=os.path.join(self.package_folder, "lib", "cmake", "ios-cmake"),
+             keep_path=False)
         self._chmod_plus_x(os.path.join(self.package_folder, "bin", "cmake-wrapper"))
 
-        self.copy("LICENSE.md", dst="licenses", src=self._source_subfolder, keep_path=False)
+        copy(self, "LICENSE.md",
+             dst=os.path.join(self.package_folder, "licenses"),
+             src=self.source_folder,
+             keep_path=False)
         # satisfy KB-H014 (header_only recipes require headers)
-        tools.save(os.path.join(self.package_folder, "include", "dummy_header.h"), "\n")
+        save(self, os.path.join(self.package_folder, "include", "dummy_header.h"), "\n")
```

Comment:
This is `package_type = "build-script"`, this hook workaround shoulnd't be needed right?